lib/logger/lib/logger.ex

```diff
@@ -573,7 +573,7 @@ defmodule Logger do
     if compare_levels(level, min_level) != :lt do
       macro_log(level, data, metadata, caller)
     else
-      :ok
+      handle_unused_variable_warnings(data)
```

Comment:
What if we have something like `Logger.info msg` where `msg` is a function with arity 0? The function will be called when it should not, right? I think you will need to validate the bindings somewhere, somehow.